pandas/tests/indexes/interval/test_base.py

```diff
@@ -12,7 +13,7 @@ class TestBase(Base):
     in test_interval.py or the specific test file (e.g. test_astype.py)
     """
 
-    _holder = IntervalIndex
+    _holder = IntervalIndex # type: Optional[Type[IntervalIndex]]
```

Comment:
Mypy considers the initial assignment as the definition of a variable. If you do not explicitly specify the type of the variable, mypy infers the type based on the static type of the value expression see https://mypy.readthedocs.io/en/latest/type_inference_and_annotations.html#type-inference

so to fix the error reported by mypy here, the explicit type declaration needs to be added in the `Base` class

Reply:
Ah, thanks for the explanation! I've updated the branch to also set the type in the `Base` class.